### ISSUE TITLE
`q` term in Logit loglike

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#3782]
The most likely explanation here is that I've been staring at a screen for too long.  That said, there is a `q` term in `Logit.loglike` (and loglikeobs) that doesn't appear elsewhere.  It looks like the fact that `cdf(-x) = 1-cdf(x)` may be taking its place, but it merits double-checking.  [Logit's](https://github.com/statsmodels/statsmodels/blob/master/statsmodels/discrete/discrete_model.py#L1288) loglikeobs, score_obs, and hessian are equivalent to:

```
    def loglikeobs(self, params):
        q = 2*self.endog - 1
        exog = self.exog
        Xb = np.dot(exog, params)
        return np.log(self.cdf(q*Xb))

    def score_obs(self, params):
        endog = self.endog
        exog = self.exog
        Xb = np.dot(exog, params)
        prob = self.cdf(Xb)
        return (endog - prob)[:, None] * exog

    def hessian(self, params):
        exog = self.exog
        Xb = np.dot(exog, params)
        prob = self.cdf(Xb)
        return -np.dot(prob*(1-prob)*exog.T, exog)
```

A derivative of `loglikeobs` w/r/t `Xb` gives `q * self.pdf(q*Xb)/self.cdf(q*Xb)`.  We can simplify this using the identity <s>`pdf(x)/cdf(x) = cdf(x) * (1-cdf(x))`.  Along with the symmetry noticed above, that means we can drop most of the `q` terms and just have `q * self.cdf(Xb) * (1-self.cdf(Xb)`.  That bears a much stronger resemblance to `hessian` than it does to `score_obs`, and even then there is a `q` that has gone missing.</s>  `pdf(x) = cdf(x) * (1-cdf(x))`.  So the derivative becomes `q*(1-cdf(q*Xb))` which for `q==1` is `1 - cdf(Xb)` and for `q==-1` is (by symmetry cdf(-X) = 1-cdf(X)) `0-cdf(Xb)`, i.e. `endog-cdf(Xb)` in all cases.  So, uh, never mind.

[MNLogit.loglikeobs](https://github.com/statsmodels/statsmodels/blob/master/statsmodels/discrete/discrete_model.py#L1719) does not have the `q` term:

```
def loglikeobs(self, params):
        params = params.reshape(self.K, -1, order='F')
        Xb = np.dot(self.exog, params)
        prob = self.cdf(Xb)
        return self.wendog * np.log(prob)

def score_obs(self, params):
	params = params.reshape(self.K, -1, order='F')
	Xb = np.dot(self.exog, params)
	prob = self.cdf(Xb)
        
	fitted = prob[:, 1:]
	resid = self.wendog[:, 1:] - fitted
	scobs = resid[:, :, None] * self.exog[:, None, :]
	return scobs.reshape(self.nobs, -1)

def hessian(self, params): # Not really
      Xb = np.dot(self.exog, params)
      return (self.exog * self.pdf(Xb)).dot(self.exog)
```

(The `MNLogit.hessian` here is a stand-in for the real function.  `pdf` is not implemented, and I've ignored dimensions since it is beside the point)

In `loglikeobs` we are multiplying by `self.wendog`, so the only probability that matters is the one for the option that was realized.  In `score_obs` and `hessian`, we are not multiplying by `wendog`.  

TLDR: it isn't obvious that `Logit.score` and `Logit.hessian` represent derivatives of `logit.loglike`, nor that MNLogit functions behave like their Logit analogues.  Is there some "summing to one" magic going on that makes this all internally consistent?

**Comment by josef-pkt** [comment]
Did you notice that q is just the sign -1 or +1 for endog= 0 or 1?
q = 2*self.endog - 1

maybe this assumes symmetry of the cdf function.

**Comment by jbrockmendel** [comment]
Yes I saw that.

Using a random sample I just verified that `score_obs` matches the derivative mentioned above (after multiplying by self.exog).  So it does look like staring-at-it-too-long was the main issue.  I'll see if I can clarify the derivation and save the next reader some trouble.

The version of `MNLogit.hessian` above can't be used right now because `MNLogit.pdf` is not implemented.  Implementing `pdf` is fairly straightforward, in part because-- after my math is double-checked-- it can be very nearly copy/pasted out of `hessian`.  One nice thing about that one-liner is that `Logit.hessian` can be simplified to the same thing with the identity `Logit.pdf = Logit.cdf*(1-Logit..cdf)`

**Comment by jbrockmendel** [comment]
See long strikeout edit above.
